lib/miq_pglogical.rb

```diff
@@ -1,90 +1,58 @@
 require 'pg'
-require 'pg/pglogical'
-require 'pg/pglogical/active_record_extension'
+require 'pg/logical_replication'
 
 class MiqPglogical
   include Vmdb::Logging
 
-  REPLICATION_SET_NAME = 'miq'.freeze
-  NODE_PREFIX = "region_".freeze
+  PUBLICATION_NAME = 'miq'.freeze
   ALWAYS_EXCLUDED_TABLES = %w(ar_internal_metadata schema_migrations repl_events repl_monitor repl_nodes).freeze
 
   attr_reader :configured_excludes
 
   def initialize
-    @connection = ApplicationRecord.connection
+    @connection = ApplicationRecord.connection.raw_connection
     self.configured_excludes = provider? ? active_excludes : self.class.default_excludes
   end
 
-  # Sets the tables that should be used to create the replication set using refresh_excludes
+  # Sets the tables that should be used to create the publication using refresh_excludes
   def configured_excludes=(new_excludes)
     @configured_excludes = new_excludes | ALWAYS_EXCLUDED_TABLES
   end
 
-  # Returns the excluded tables that are currently being used by pglogical
+  # Returns the excluded tables that are currently being used
   # @return Array<String> the table list
   def active_excludes
     return [] unless provider?
-    @connection.tables - included_tables
+    ApplicationRecord.connection.tables - included_tables
   end
 
-  # Returns whether or not this server is configured as a provider node
-  # @return Boolean
   def provider?
-    pglogical.enabled? && pglogical.replication_sets.include?(REPLICATION_SET_NAME)
+    pglogical.publications.map { |p| p["name"] }.include?(PUBLICATION_NAME)
```

Comment:
This feels like it should move into the pg-logical_replication gem with publication_name as a parameter, and the logic for knowing about mapping publication names is something the gem should be concerned about. I could see...

```ruby
# gem - client class
def provider_for?(publication_name)
  publications.map { |p| p["name"] }.include?(publication_name)
end

# here
def provider?
  pglogical.provider_for?(PUBLICATION_NAME)
end
```

Reply:
Also, it might be slightly more efficient to do

```ruby
pglogical.publications.any? { |p| p["name"] == PUBLICATION_NAME) }
```

Reply:
If I move this into the gem I'd like to get away from the `provider` language and call the method something like `has_publication?`. Naming is hard though.

Reply:
Maybe `publishes?`